warehouse/observations/tasks.py

```diff
@@ -149,3 +154,127 @@ def report_observation_to_helpscout(task, request: Request, model_id: UUID) -> N
     model.additional["helpscout_conversation_url"] = new_convo_location
     request.db.add(model)
     return
+
+
+@tasks.task(
+    bind=True,
+    ignore_result=True,
+    acks_late=True,
+    autoretry_for=(RequestException,),
+    retry_backoff=True,
+)
+def auto_quarantine_project(task, request: Request, model_id: UUID) -> None:
+    """
+    Automatically quarantine a Project for Admin review.
+
+    Conditions must be met:
+
+    - ObservationKind is IsMalware
+    - Observed Project is not already quarantined
+    - Observed Project has at least 2 Observations, at least 1 by `User.is_observer`
+    """
+    # Fetch the Observation from the database, load the related Project
+    model = request.db.get(Observation, model_id)
```

Comment:
would `observation` and `observation_id` maybe be a bit clearer? especially when referenced later in this function.